src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -375,12 +375,19 @@ loop_inst::concatenated_memory_mapping::ptr loop_inst::create_concat_memory_map(
     if (extern_mem_ptr != nullptr) {
         layout sliced_layout = intern_prim->get_output_layout(internal_id.idx);
         auto inter_mem_ptr = intern_prim->output_memory_ptr(internal_id.idx);
-        if (inter_mem_ptr == nullptr) {
+        if (inter_mem_ptr == nullptr || shape_changed()) {
             // if inner body intern_prim has no output memory because it has dynamic shape,
             // calculate inner body intern_prim layout using concat_mem's layout.
             auto updated_sliced_layout = sliced_layout.get_partial_shape();
             OPENVINO_ASSERT(updated_sliced_layout[io_prim_map.axis].is_static() || num_iterations > 0,
                                     "Not allowed dynamic dimension for axis when num_iteraiont is negative");
+
+            auto dyn_sliced_layout = dynamic_sliced_layout_mappings.find(internal_id.pid);
+            if (dyn_sliced_layout != dynamic_sliced_layout_mappings.end() &&
+                dyn_sliced_layout->second.is_dynamic()) {
+                updated_sliced_layout = dyn_sliced_layout->second;
+            }
```

Comment:
What is the target case?  Seems that outer mem is allocated, which means it is either of static shape or "bounded dynamic shape". Which case?

Reply:
I am curious because why we need this dyn_sliced_layout mapping. Seems that it just holds original dynamic shape. Which means that it can be pulled from layout of program_node.

Reply:
The target case: 
-shape "chunks[-1, 1, -1]" -data_shape "chunks[32, 1, 80000][19, 1, 80000]"
Multiple infer with different shape is required for loop operation

In case sliced layout of loop is dynamic, the static shape is decided during first infer execution. But during 2nd infer, the sliced layout of loop is not updated. To update the sliced layout, the original dynamic shape info is needed at loop_inst::create_concat_memory_map().

Reply:
I agree with @yeonbok 's comment.
I think if you get the output layout from the program node, you don't need to use dynamic_sliced_layout_mappings to handle the original output pshape.

Reply:
Updated to get the output layout from program node instead of saving dynamic shape